source/common/http/codec_client.cc

```diff
@@ -138,6 +138,19 @@ void CodecClient::onData(Buffer::Instance& data) {
       host_->cluster().stats().upstream_cx_protocol_error_.inc();
     }
   }
+
+  // The HTTP/1 codec will pause dispatch after a single response is complete. If there is more
+  // data, drain and close the connection.
+  if (codec_->protocol() < Protocol::Http2 && data.length() > 0) {
```

Comment:
I occurs to me that this check could be done in an HTTP1 ClientConnectionImpl::dispatch override which verifies that data is empty when attempting to return success.  What do you think?

Reply:
So much cleaner. Gets uniformly handled at the level of codec client.